hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBootstrapRelation.scala

```diff
@@ -107,37 +107,35 @@ class HoodieBootstrapRelation(@transient val _sqlContext: SQLContext,
     })
 
     // Prepare readers for reading data file and skeleton files
-    val dataReadFunction = new ParquetFileFormat()
-        .buildReaderWithPartitionValues(
-          sparkSession = _sqlContext.sparkSession,
-          dataSchema = dataSchema,
-          partitionSchema = StructType(Seq.empty),
-          requiredSchema = requiredDataSchema,
-          filters = if (requiredSkeletonSchema.isEmpty) filters else Seq() ,
-          options = Map.empty,
-          hadoopConf = _sqlContext.sparkSession.sessionState.newHadoopConf()
-        )
-
-    val skeletonReadFunction = new ParquetFileFormat()
-      .buildReaderWithPartitionValues(
-        sparkSession = _sqlContext.sparkSession,
-        dataSchema = skeletonSchema,
-        partitionSchema = StructType(Seq.empty),
-        requiredSchema = requiredSkeletonSchema,
-        filters = if (requiredDataSchema.isEmpty) filters else Seq(),
-        options = Map.empty,
-        hadoopConf = _sqlContext.sparkSession.sessionState.newHadoopConf()
-      )
-
-    val regularReadFunction = new ParquetFileFormat()
-      .buildReaderWithPartitionValues(
-        sparkSession = _sqlContext.sparkSession,
-        dataSchema = fullSchema,
-        partitionSchema = StructType(Seq.empty),
-        requiredSchema = requiredColsSchema,
-        filters = filters,
-        options = Map.empty,
-        hadoopConf = _sqlContext.sparkSession.sessionState.newHadoopConf())
+    val dataReadFunction = HoodieDataSourceHelper.buildHoodieParquetReader(
+      sparkSession = _sqlContext.sparkSession,
```

Comment:
legacy codes. i think that's because the parent class `BaseRelation` requires to implement abstract method `sqlContext`, and here mark it as `@transient`.